roles.cpp

```diff
@@ -0,0 +1,163 @@
+////////////////////////////////////////////////////
+// Automatically generated by "generate-roles.go" //
+////////////////////////////////////////////////////
+
```

Comment:
This is mostly a matter of style, if we follow Google's code style then it's not needed.